gnomad/utils/vcf.py

```diff
@@ -92,3 +339,488 @@ def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpressi
+                        "Number": "A",
+                        "Description": f"Filtering allele frequency (using Poisson 99% CI) {make_combo_header_text('for', group_types, combo_fields, prefix)}",
+                    },
+                }
+            info_dict.update(combo_dict)
+    return info_dict
+
+
+def add_as_info_dict(INFO_DICT: Dict[str, Dict[str, str]]) -> Dict[str, Dict[str, str]]:
+    """
+    Updates info dictionary with allele-specific terms and their descriptions.
+
+    Used in VCF export.
+
+    :param INFO_DICT: Dictionary containing site-level annotations and their descriptions.
+    :return: Dictionary with allele specific annotations, their descriptions, and their VCF number field.
+    """
+    prefix_text = "Allele-specific"
+    AS_DICT = {}
+
+    for field in AS_FIELDS:
+        # Strip AS_ from field name
+        site_field = field[3:]
+
+        AS_DICT[field] = {}
+        AS_DICT[field]["Number"] = "A"
+        AS_DICT[field][
+            "Description"
+        ] = f"{prefix_text} {INFO_DICT[site_field]['Description'][0].lower()}{INFO_DICT[site_field]['Description'][1:]}"
+
+    return AS_DICT
+
+
+def make_vcf_filter_dict(
+    snp_cutoff: float, indel_cutoff: float, inbreeding_cutoff: float
+) -> Dict[str, str]:
+    """
+    Generates dictionary of Number and Description attributes to be used in the VCF header, specifically for FILTER annotations.
+
+    :param ht: Table containing global annotations of the Random Forests SNP and indel cutoffs.
+    :param snp_cutoff: Minimum SNP cutoff score from random forest model.
+    :param indel_cutoff: Minimum indel cutoff score from random forest model.
+    :param inbreeding_cutoff: Inbreeding coefficient hard cutoff.
+    :return: Dictionary keyed by VCF FILTER annotations, where values are Dictionaries of Number and Description attributes.
+    """
+    filter_dict = {
+        "AC0": {
+            "Description": "Allele count is zero after filtering out low-confidence genotypes (GQ < 20; DP < 10; and AB < 0.2 for het calls)"
+        },
+        "InbreedingCoeff": {"Description": f"InbreedingCoeff < {inbreeding_cutoff}"},
+        "RF": {
+            "Description": f"Failed random forest filtering thresholds of {snp_cutoff}, {indel_cutoff} (probabilities of being a true positive variant) for SNPs, indels"
+        },
+        "PASS": {"Description": "Passed all variant filters"},
+    }
+    return filter_dict
+
+
+def make_hist_bin_edges_expr(ht: hl.Table, prefix: str) -> Dict[str, str]:
+    """
+    Create dictionaries containing variant histogram annotations and their associated bin edges, formatted into a string
+    separated by pipe delimiters.
+
+    :param ht: Table containing histogram variant annotations.
+    :param prefix: Prefix text for age histogram bin edges. 
+    :return: Dictionary keyed by histogram annotation name, with corresponding reformatted bin edges for values.
+    """
+    edges_dict = {
+        f"{prefix}het": "|".join(
```

Comment:
Is prefix expected to have and _ at the end? I think it might be better to just add an _ if prefix is not "" and set prefix default as ""